### PR TITLE
fix: check number of outputs in ingest test 

### PR DESCRIPTION
**Pull request by ryannikolaidis**
In CI the "check number of outputs" ingest tests are not properly evaluated because the syntax is not recognized. Example [here]( https://github.com/Unstructured-IO/unstructured/actions/runs/5971636507/job/16201013597#step:7:1818) where you'll see the check log a warning 

```
/home/runner/work/unstructured/unstructured/test_unstructured_ingest/check-num-files-expected-output.sh: 19: [[: not found
```

 This PR updates the syntax such that it properly evaluate the condition. Example of the condition properly triggering [here](https://github.com/Unstructured-IO/unstructured/actions/runs/5986027131/job/16238759785).

### CODE REVIEW

File: test_unstructured_ingest/check-num-dirs-output.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
 
 NUMBER_OF_FOUND_DIRS="$(find "$OUTPUT_DIR" -type d -exec printf '.' \; | wc -c | xargs)"
 
-if [ "$NUMBER_OF_FOUND_DIRS" != "$EXPECTED_NUM_DIRS" ]; then
+if [ "$NUMBER_OF_FOUND_DIRS" -ne "$EXPECTED_NUM_DIRS" ]; then
```
Comment (cragwolfe) [inline test_unstructured_ingest/check-num-dirs-output.sh:19]: worth adding a NOTE before each condition with a warning not to change to double-bracket syntax due to CI issues?

Comment (ryannikolaidis) [reply test_unstructured_ingest/check-num-dirs-output.sh:19]: yep, done!

File: test_unstructured_ingest/test-ingest-local.sh [hunk 2]
```diff
@@ -17,4 +17,4 @@ PYTHONPATH=. ./unstructured/ingest/main.py \
     --file-glob "*.html" \
     --input-path example-docs
 
-sh "$SCRIPT_DIR"/check-num-files-output.sh 11 $OUTPUT_FOLDER_NAME
+sh "$SCRIPT_DIR"/check-num-files-output.sh 12 $OUTPUT_FOLDER_NAME
```
Comment (ryannikolaidis) [inline test_unstructured_ingest/test-ingest-local.sh:20]: was not caught before due to this issue with conditional

**Review by cragwolfe — APPROVED** [review]
🙏

**Pull request merged**